vochain/indexer/indexer.go

```diff
@@ -344,25 +341,38 @@ func (idx *Indexer) AfterSyncBootstrap() {
 	// Block the new votes addition until the recovery finishes.
 	idx.recoveryBootLock.Lock()
 	defer idx.recoveryBootLock.Unlock()
-	// Find those processes which do not have yet final results,
-	// they are considered live so we need to compute the temporary
-	// results (or only its weight in case of Encrypted)
-	//prcs, err := idx.ProcessList(nil, 0, 0, "", 0, "", "READY", true)
-	prcs := [][]byte{}
-	err := idx.db.ForEach(
-		badgerhold.Where("FinalResults").Eq(false),
-		func(p *indexertypes.Process) error {
-			prcs = append(prcs, p.ID)
-			return nil
-		})
 
+	queries, ctx, cancel := idx.timeoutQueries()
+	defer cancel()
+
+	prcIDs, err := queries.GetProcessIDsByFinalResults(ctx, false)
 	if err != nil {
 		log.Error(err)
 	}
-	log.Infof("recovered %d live results processes", len(prcs))
+
+	if !enableBadgerhold {
+		// Find those processes which do not have yet final results,
+		// they are considered live so we need to compute the temporary
+		// results (or only its weight in case of Encrypted)
+		//prcs, err := idx.ProcessList(nil, 0, 0, "", 0, "", "READY", true)
```

Comment:
This commented code was here to help implement the SQL migration, since querying ProcessList with the state=READY would have the same output as your custom method `GetProcessIDsByFinalResults`. However, your solution is better, so you can remove this comment.